src/AppBundle/Helper/DeckValidationHelper.php

```diff
@@ -152,6 +157,19 @@ public function canIncludeCard($deck, $slot, $deck_options = []) {
 			}
 		}
 
+		// always allow the required cards regardless
+		$requirements = $investigator->getDeckRequirements();
+		if ($requirements) {
+			$parsed = $this->parseReqString($requirements);
+			if ($parsed && $parsed['card'] && $parsed['card'][$card->getCode()]) {
+				return true;
+			}
+		}
+
+		if ($card->getDeckLimit() > 0 && is_null($card->getXp())) {

```

Comment:
This is a rather permissive check for "anything that can go in the deck and does not have a level". This covers story assets, weaknesses, and signatures. Should be fine, given that restrictions are accounted for above.